web/client/components/catalog/RecordItem.jsx

```diff
@@ -135,47 +137,77 @@ class RecordItem extends React.Component {
         return crs && !isSRSAllowed(crs);
     }
 
+    onAddToMap = (record, serviceType = record.serviceType) => {
+        if (this.isSRSNotAllowed(record)) {
+            return this.props.onError('catalog.srs_not_allowed');
+        }
+        this.setState({ loading: true });
+        return API[serviceType].getLayerFromRecord(record, {
```

Comment:
why this has been moved to the component?

is this part of logic also needed in other catalogs around mapstore, outside the viewer?

Reply:
It was already a part of the component, I merely added a function block to make the component cleaner. Only in viewer (geostory, context, map) except dashboard catalog which is handled in a separate issue #10303